mlir/lib/Dialect/Rock/utility/transformMapUtils.cpp

```diff
@@ -46,15 +47,41 @@ mlir::rock::untransform(OpBuilder &b, Value transformed, ArrayAttr existing) {
     transformList.push_back(transform.getTransform());
     ret = transform.getInput();
   }
-  return {ret, b.getArrayAttr(transformList)};
+
+  bool isBig = llvm::any_of(transformList, [](Attribute a) {
+    return needs64BitIndices(cast<TransformMapAttr>(a));
+  });
+  auto bufferType = cast<ShapedType>(ret.getType());
+  int64_t elemBytes;
+  if (auto shapedTy = dyn_cast<ShapedType>(bufferType.getElementType()))
+    elemBytes =
+        (shapedTy.getNumElements() * shapedTy.getElementTypeBitWidth()) / 8;
+  else
+    elemBytes = bufferType.getElementTypeBitWidth() / 8;
+
+  if (bufferType.hasStaticShape())
+    isBig = (bufferType.getNumElements() * elemBytes) >=
```

Comment:
this code was elsewhere as well.. might worth factoring to a general util

Reply:
Yeah, that did need to be a utility, fixed.